backend/seafile/seafile.go

```diff
@@ -0,0 +1,1132 @@
+	dstDir, dstFilename := path.Split(dstPath)
+
+	// We have to make sure the destination path exists on the server
+	err = f.mkMultiDir(ctx, dstLibraryID, dstDir)
+	if err != nil {
+		return nil, err
+	}
+
+	fs.Debugf(nil, "%s: Move [%s]%s to [%s]%s", f.Name(), srcLibraryName, srcPath, dstLibraryName, dstDir)
+	op, err := f.moveFile(ctx, srcLibraryID, srcPath, dstLibraryID, dstDir)
+	if err != nil {
+		return nil, err
+	}
+
+	if op.Name != dstFilename {
+		// Seafile seems to be acting strangely if the renamed file already exists (some cache issue maybe?)
+		// It's better to delete the destination if it already exists
+		fileDetail, err := f.getFileDetails(ctx, dstLibraryID, dstPath)
+		if err != nil && err != fs.ErrorObjectNotFound {
+			return nil, err
+		}
+		if fileDetail != nil {
+			fs.Debugf(nil, "%s: Delete existing destination file '%s'", f.Name(), dstPath)
+			err = f.deleteFile(ctx, dstLibraryID, dstPath)
+			if err != nil {
+				return nil, err
+			}
+		}
+		fs.Debugf(nil, "%s: Rename '%s' to '%s' (in '%s')", f.Name(), op.Name, dstFilename, dstDir)
+		_, err = f.renameFile(ctx, dstLibraryID, path.Join(dstDir, op.Name), dstFilename)
+		if err != nil {
+			return nil, err
+		}
+	}
+	// Create a new object from the result
+	return f.NewObject(ctx, remote)
+}
+
+// ==================== Optional Interface fs.Purger ====================
+
+// Purge all files in the root and the root directory
+//
+// Implement this if you have a way of deleting all the files
+// quicker than just running Remove() on the result of List()
+//
+// Return an error if it doesn't exist
+func (f *Fs) Purge(ctx context.Context) error {
+	if f.libraryName == "" {
+		return errors.New("Cannot delete from the root of the server. Please select a library")
+	}
+	libraryID, err := f.getLibraryID(ctx, f.libraryName)
+	if err != nil {
+		return err
+	}
+	if f.rootDirectory == "" {
+		// Delete library
+		err = f.deleteLibrary(ctx, libraryID)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+	err = f.deleteDir(ctx, libraryID, f.rootDirectory)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// ==================== Optional Interface fs.CleanUpper ====================
+
+// CleanUp the trash in the Fs
+func (f *Fs) CleanUp(ctx context.Context) error {
+	if f.libraryName == "" {
+		return errors.New("Cannot clean up at the root of the seafile server: please select a library to clean up")
+	}
+	libraryID, err := f.getLibraryID(ctx, f.libraryName)
+	if err != nil {
+		return err
+	}
+	return f.emptyLibraryTrash(ctx, libraryID)
+}
+
+// ==================== Optional Interface fs.Abouter ====================
+
+// About gets quota information
+func (f *Fs) About(ctx context.Context) (usage *fs.Usage, err error) {
+	accountInfo, err := f.getUserAccountInfo(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	usage = &fs.Usage{
+		Used: fs.NewUsageValue(accountInfo.Usage), // bytes in use
+	}
+	if accountInfo.Total > 0 {
```

Comment:
Does `Total <=0` here mean that there is no quota?

Reply:
This is right, by default a user has no quota and it's not returning the size of the disk (returning -2 for no quota)